src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/api/controller/SearchController.kt

```diff
@@ -10,34 +10,45 @@ import uk.gov.justice.digital.hmpps.personrecord.api.model.PersonIdentifierRecor
 import uk.gov.justice.digital.hmpps.personrecord.jpa.entity.PersonEntity
 import uk.gov.justice.digital.hmpps.personrecord.jpa.repository.PersonRepository
 import uk.gov.justice.digital.hmpps.personrecord.model.types.SourceSystemType
+import uk.gov.justice.digital.hmpps.personrecord.service.EventKeys
+import uk.gov.justice.digital.hmpps.personrecord.service.TelemetryService
+import uk.gov.justice.digital.hmpps.personrecord.service.type.TelemetryEventType
 
 @RestController
 @PreAuthorize("hasRole('ROLE_CORE_PERSON_RECORD_API__SEARCH__RO')")
 class SearchController(
+  private val telemetryService: TelemetryService,
   private val personRepository: PersonRepository,
 ) {
 
   @GetMapping("/search/offender/{crn}")
   fun searchByCrn(@NotBlank @PathVariable(name = "crn") crn: String): List<PersonIdentifierRecord> {
-    return handlePersonRecord(personRepository.findByCrn(crn), crn)
+    return handlePersonRecord(personRepository.findByCrn(crn), Pair(EventKeys.CRN, crn))
   }
 
   @GetMapping("/search/prisoner/{prisonNumber}")
   fun searchByPrisonNumber(@NotBlank @PathVariable(name = "prisonNumber") prisonNumber: String): List<PersonIdentifierRecord> {
-    return handlePersonRecord(personRepository.findByPrisonNumberAndSourceSystem(prisonNumber), prisonNumber)
+    return handlePersonRecord(personRepository.findByPrisonNumberAndSourceSystem(prisonNumber), Pair(EventKeys.PRISON_NUMBER, prisonNumber))
   }
 
   @GetMapping("/search/defendant/{defendantId}")
   fun searchByDefendantId(@NotBlank @PathVariable(name = "defendantId") defendantId: String): List<PersonIdentifierRecord> {
-    return handlePersonRecord(personRepository.findByDefendantId(defendantId), defendantId)
+    return handlePersonRecord(personRepository.findByDefendantId(defendantId), Pair(EventKeys.DEFENDANT_ID, defendantId))
   }
 
-  private fun handlePersonRecord(personEntity: PersonEntity?, identifier: String): List<PersonIdentifierRecord> = when {
-    personEntity != PersonEntity.empty -> buildListOfLinkedRecords(personEntity!!)
-    else -> throw PersonRecordNotFoundException(identifier)
+  private fun handlePersonRecord(personEntity: PersonEntity?, identifier: Pair<EventKeys, String>): List<PersonIdentifierRecord> = when {
+    personEntity != PersonEntity.empty -> buildListOfLinkedRecords(personEntity!!, identifier)
+    else -> throw PersonRecordNotFoundException(identifier.second)
   }
 
-  private fun buildListOfLinkedRecords(personEntity: PersonEntity): List<PersonIdentifierRecord> {
+  private fun buildListOfLinkedRecords(personEntity: PersonEntity, identifier: Pair<EventKeys, String>): List<PersonIdentifierRecord> {
+    telemetryService.trackEvent(
+      TelemetryEventType.CPR_SEARCH_REQUEST,
```

Comment:
What does this give us over a simple appInsights query on `requests`?
I'm not sure this adds any value

Reply:
Only value is seeing what UUID is returned as part of the request.

Reply:
In the case that we dont want to be raising any excess app insights events. I'll close the PR and we can just query to find the UUID from the record

Reply:
if it's useful to be able to find out what UUID is returned then fair enough